api/data_refinery_api/views.py

```diff
@@ -511,21 +511,42 @@ def perform_update(self, serializer):
                         except Exception:
                             city = "COULD_NOT_DETERMINE"
 
-                        new_user_text = "New user " + supplied_email_address + " from " + city + " [" + remote_ip + "] downloaded a dataset! (" + str(old_object.id) + ")"
-                        webhook_url = settings.ENGAGEMENTBOT_WEBHOOK
-                        slack_json = {
-                            "channel": "ccdl-general", # Move to robots when we get sick of these
-                            "username": "EngagementBot",
-                            "icon_emoji": ":halal:",
-                            "attachments":[
-                                {   "color": "good",
-                                    "text": new_user_text
-                                }
-                            ]
-                        }
+                        user_agent = self.request.META.get('HTTP_USER_AGENT', None)
+                        total_samples = len(set([accession_code for experiment in new_data.values() for accession_code in experiment]))
                         response = requests.post(
-                            webhook_url,
-                            json=slack_json,
+                            settings.ENGAGEMENTBOT_WEBHOOK,
+                            json={
+                                'channel': 'ccdl-general', # Move to robots when we get sick of these
+                                'username': 'EngagementBot',
+                                'icon_emoji': ':halal:',
+                                'attachments': [
+                                    {
+                                        'color': 'good',
+                                        'title': 'New dataset download',
+                                        'fallback': 'New dataset download',
+                                        'title_link': 'http://www.refine.bio/dataset/' + str(old_object.id),
```

Comment:
Can we do string concat like this instead?

```"http://www.refine.bio/dataset/{0}".format(old_object.id)```

Reply:
```suggestion
                                        'title_link': 'http://www.refine.bio/dataset/'.format(old_object.id),
```